lib/blocklist.js

```diff
@@ -0,0 +1,48 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+/**
+ * This module checks to see if an ip address is contained in the blocklist. This blocklist
+ * is loaded from a CSV file that contains a column of ip addresses.
+ *
+ * Currently, this file is obtained from Spamhaus BCL. After signing up with
+ * an account you should be able to run /scripts/spamhaus-sync.sh to obtain list.
+ *
+ */
+var Promise = require('bluebird')
+var readFile = Promise.promisify(require('fs').readFile)
+var parse = Promise.promisify(require('csv-parse'))
+var path = require('path')
+
+// Contains ip addresses in blocklist
+var blocklist
+
+module.exports = {
+  load: function (csvPath) {
+    csvPath = path.join(__dirname, csvPath)
+
+    return readFile(csvPath, 'utf8')
+      .then(function (data) {
+        return parse(data)
+      })
+      .then(function (rows) {
+        // Convert array to map for faster lookup
+        var ips = {}
+        rows.forEach(function (row) {
+          ips[row] = {}
+        })
+        blocklist = ips
+      })
+  },
+  clear: function () {
+    blocklist = undefined
+  },
+  contains: function (ip) {
+    if (!blocklist) {
+      // Throw error if blocklist has not been loaded
+      throw new Error('Please load blocklist')
```

Comment:
Might need to adjust this. Is throwing here risky in terms of breaking a lot of requests? Should the blocklist component be more isolated (if it is not loaded, log the fatal / error but proceed with the customs request)?